builtin/logical/transit/simple_crud.go

```diff
@@ -0,0 +1,90 @@
+package transit
+
+import (
+	"sync"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+// Directly implements CRUD operations without caching, mapped to the backend,
+// but implements shared locking to ensure that we can't overwrite data on the
+// backend from multiple operators
+type simplePolicyCRUD struct {
+	sync.RWMutex
+	locks         map[string]*sync.RWMutex
+	locksMapMutex sync.RWMutex
+}
+
+func newSimplePolicyCRUD() *simplePolicyCRUD {
+	return &simplePolicyCRUD{
+		locks: map[string]*sync.RWMutex{},
+	}
+}
+
+func (p *simplePolicyCRUD) ensureLockExists(name string) {
+	p.locksMapMutex.RLock()
+
+	if p.locks[name] == nil {
+		p.locksMapMutex.RUnlock()
+		p.locksMapMutex.Lock()
+		// Make sure nothing has appeared since we switched the lock type
+		if p.locks[name] == nil {
+			p.locks[name] = &sync.RWMutex{}
+		}
+		p.locksMapMutex.Unlock()
```

Comment:
It can. There isn't any benefit to doing so, however, since there is no access to any other resource before the function returns. So either way works.